src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,418 @@
+
+        posix_time = attrs.get("MeasurementDatePosix")
+        nano_secs = attrs.get("MeasurementDateNanoSecs", 0)
+
+        if posix_time is not None:
+            # Return as UTC
+            return datetime.fromtimestamp(posix_time + nano_secs / 1e9, tz=timezone.utc)
+        return None
+
+    def _get_experimenter_info(self) -> Dict[str, str]:
+        """Get experimenter information."""
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+
+        return {
+            "username": self._decode_string(attrs.get("ExperimenterUsername", [])),
+            "setup_id": self._decode_string(attrs.get("ExperimenterSetupID", [])),
+            "hostname": self._decode_string(attrs.get("ExperimenterHostname", [])),
+        }
+
+    def _get_geometric_transformations(self) -> Dict[str, np.ndarray]:
+        """
+        Get geometric transformations: translation, rotation, and labeling origin.
+
+        Returns
+        -------
+        dict with keys 'translation', 'rotation', 'labeling_origin'
+        """
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+        return {
+            "translation": attrs.get("GeomTransTransl"),
+            "rotation": attrs.get("GeomTransRot"),
+            "labeling_origin": attrs.get("LabelingOriginTransl"),
+        }
+
+    def _get_mesc_version_info(self) -> Dict[str, Any]:
+        """Get MESc (Measurement Session Container) software version information."""
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+
+        return {
+            "version": self._decode_string(attrs.get("CreatingMEScVersion", [])),
+            "revision": attrs.get("CreatingMEScRevision"),
+        }
+
+    def _get_pmt_settings(self) -> Dict[str, Dict[str, float]]:
+        """Get photomultiplier tube (PMT) settings if available from XML metadata."""
+        try:
+            attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+
+            xml_data = attrs.get("MeasurementParamsXML")
+            if xml_data is None:
+                return {}
+
+            xml_str = xml_data.decode("latin-1")
+
+            root = ET.fromstring(xml_str)
+
+            pmt_settings = {}
+            for gear in root.findall('.//Gear[@type="PMT"]'):
+                try:
+                    channel_name = gear.find('.//param[@name="channel_name"]').get("value")
+                    voltage = float(gear.find('.//param[@name="reference_voltage"]').get("value"))
+                    warmup = float(gear.find('.//param[@name="warmup_time"]').get("value"))
+                    pmt_settings[channel_name] = {"voltage": voltage, "warmup_time": warmup}
+                except (AttributeError, ValueError):
+                    continue
+            return pmt_settings
+        except Exception:
+            return {}
+
+    def _get_scan_parameters(self) -> Dict[str, Any]:
```

Comment:
```suggestion
    def _get_scan_parameters(self) -> dict[str, Any]:
```
You can also use it here like this.